app/components/files/index.ts

```diff
@@ -36,22 +37,8 @@ const filesLocalPathValidation = async (files: FileModel[], authorId: string) =>
 };
 
 const enhance = withObservables(['post'], ({database, post}: EnhanceProps) => {
-    const config = observeConfig(database);
-    const enableMobileFileDownload = config.pipe(
-        switchMap((cfg) => of$(cfg?.EnableMobileFileDownload !== 'false')),
-    );
-
-    const publicLinkEnabled = config.pipe(
-        switchMap((cfg) => of$(cfg?.EnablePublicLink !== 'false')),
-    );
-
-    const complianceDisabled = observeLicense(database).pipe(
-        switchMap((lcs) => of$(lcs?.IsLicensed === 'false' || lcs?.Compliance === 'false')),
-    );
-
-    const canDownloadFiles = combineLatest([enableMobileFileDownload, complianceDisabled]).pipe(
-        map(([download, compliance]) => compliance || download),
-    );
+    const canDownloadFiles = observeCanDownloadFiles(database);
+    const publicLinkEnabled = observeConfigBooleanValue(database, 'EnablePublicLink');
```

Comment:
Here we are not doing `=== 'true'` but `!== 'false'`. What is the difference? When the license is undefined (or that value is undefined).

So... for example, a server without that config value set, with the old version, it would do the following:
- `cfg?.EnablePublicLink` is `undefined`
- `undefined` is different of `'false'`
- `publicLinkEnabled` is `true`

With the new version:
- `cfg?.EnablePublicLink` is `undefined`
- `undefined` is NOT equal to `'true'`
- `publicLinkEnabled` is `false`

One solution is passing the default value to this function, and handle that in the `observeConfigBooleanValue`, but should be handled with care in every place you substitute this.

Reply:
I understand what you are saying and that means there are other places where I need to make changes in this PR. 

For instance. https://github.com/mattermost/mattermost-mobile/pull/6483/files/af3664e3fb176104cd567bafd3ac60188a12bfb4#diff-2fd739f501fb0b1e8cd8c8a45af96276780a0980639e125e182cdd98af1278b0R23-R24

Reply:
Or maybe not. I think what I'll do is revert all the variables that were checking for !== false and keep that format instead of getting the useing observeConfigBooleanValue() Does that sound good? We could also add a observeConfigBooleanValueNotFalse()?

Reply:
The best approach is to pass a default value. That way the check is something like...

```
if (license?.x == null /* or license?.x === undefined */) {
   return defaultValue;
}
return license.x === 'true';
```